File: C#/백준/Bronze/11970. Fence Painting/Fence Painting.cs
using System;

namespace q{
    class w{
        static void Main(){
            int[] a = Array.ConvertAll(Console.ReadLine().Split(),int.Parse);
            int[] b = Array.ConvertAll(Console.ReadLine().Split(),int.Parse);
            if(a[1]<=b[0] || b[1]<=a[0]){
                Console.WriteLine(a[1]-a[0]+b[1]-b[0]);
            }
            else{
                int[] c = a.Concat(b).ToArray();
                Console.WriteLine(c.Max()-c.Min());
            }
        }
    }
}

File: Python/백준/Bronze/11970. Fence Painting/Fence Painting.py
aa = list(map(int,input().split()))
bb = list(map(int,input().split()))
cc = [i for i in range(min(aa),max(aa)+1)]
dd = [i for i in range(min(bb),max(bb)+1)]
ck=0
for i in cc:
    if i in dd:
        ck=1
        break
if(ck==0):
    print(max(aa)-min(aa)+max(bb)-min(bb))
else:
    cc = aa+bb
    print(max(cc)-min(cc))